File: python/guidos-gorgeous-lasagna/lasagna.py
"""lasagna.py - methods and constants to solve the exercism.io
    "Guido's Gorgeous Lasagna" problem.
"""
EXPECTED_BAKE_TIME = 40  # Expected time to bake in minutes
PREPARATION_TIME = 2  # Preparation time for one layer in minutes


def bake_time_remaining(elapsed_bake_time):
    '''
    :param elapsed_bake_time: int baking time already elapsed
    :return: int remaining bake time derived from 'EXPECTED_BAKE_TIME'

    Function that takes the actual minutes the lasagna has been in the oven as
    an argument and returns how many minutes the lasagna still needs to bake
    based on the `EXPECTED_BAKE_TIME`.
    '''

    return EXPECTED_BAKE_TIME - elapsed_bake_time


def preparation_time_in_minutes(number_of_layers):
    """preparation_time_in_minutes: Calculate the amount of time to prepare a
    lasagne with number_of_layers.

    Args:
        number_of_layers (int): The number of layers to use in calculating
        preparation_time_in_minutes.

    Returns:
        int: preparation_time_in_minutes = number_of_layers * PREPARATION_TIME
    """
    return number_of_layers * PREPARATION_TIME


def elapsed_time_in_minutes(number_of_layers, elapsed_bake_time):
    """elapsed_time_in_minutes - The total time in minutes passed since the
    begining of preparation.

    Args:
        number_of_layers (int): The number of layers to calculate preparation
        time with.
        elapsed_bake_time (int): The amount of time (in minutes) we have spent
        baking.

    Returns:
        int: The sum of the preparation_time_in_minutes + elapsed_bake_time
        (in minutes).
    """
    return preparation_time_in_minutes(number_of_layers) + elapsed_bake_time

File: java/lasagna/Lasagna.java
public class Lasagna {
    static int TIME_IN_OVEN = 40;
    static int LAYER_PREP_TIME = 2;

    /**
     * Returns the value of the constant TIME_IN_OVEN.
     * 
     * @return int: TIME_IN_OVEN
     */
    public int expectedMinutesInOven() {

        return TIME_IN_OVEN;
    }

    /**
     * Returns the amount of time left to cook.
     * 
     * @param timeCooking int: The time spent cooking so far.
     * @return int: TIME_IN_OVEN - timeCooking
     */
    public int remainingMinutesInOven(int timeCooking) {

        return TIME_IN_OVEN - timeCooking;
    }

    /**
     * Returns the expected preparation time for layers number of layers.
     * 
     * @param layers int: The number of layers to prepare.
     * @return int: The total preparation time.
     */
    public int preparationTimeInMinutes(int layers) {

        return layers * LAYER_PREP_TIME;
    }

    /**
     * Calculates the total time spent making the meal.
     * 
     * @param layers      int: The number of layers to make.
     * @param timeCooking int: The amount of time spent cooking so far.
     * @return int: Total time currently spent on making the meal.
     */
    public int totalTimeInMinutes(int layers, int timeCooking) {

        return preparationTimeInMinutes(layers) + timeCooking;
    }
}
